app/vmalert/remotewrite/remotewrite.go

```diff
@@ -27,6 +27,15 @@ var (
 	retryMaxTime      = flag.Duration("remoteWrite.retryMaxTime", time.Second*30, "The max time spent on retry attempts for the failed remote-write request. Change this value if it is expected for remoteWrite.url to be unreachable for more than -remoteWrite.retryMaxTime. See also -remoteWrite.retryMinInterval")
 )
 
+// RWClient interface wraps methods for remote write client
+type RWClient interface {
```

Comment:
Simplifying DebugRW would also make this interface much simpler.

Reply:
Yes, it does.
I suppose there won't be too many input series and rules to process in unittest, single request could handle that.
We can implement concurrency or batch if needed in the future.